bdk_chain/src/lib.rs

```diff
@@ -37,13 +37,15 @@ pub mod collections {
     #![allow(dead_code)]
     pub type HashSet<K> = alloc::collections::BTreeSet<K>;
     pub type HashMap<K, V> = alloc::collections::BTreeMap<K, V>;
+    pub use alloc::collections::btree_map::Entry;
```

Comment:
I think we should remove this.

Reply:
Not sure why though.. We can, but then we need to specify `which::Entry` at call site and feature gate it.. Not creating feature gated collection imports is what this module was intended to solve.. But maybe I am missing something.. Do you have a better way to call entry without explicit import?

Reply:
If someone wants to import both `bdk_core::collections::Entry` (which is only a `btree_map::Entry`) and `std::collections::hash_map::Entry` at the same time, it will get messy. I think having to explicitly state which `Entry` it is, is a good-to-have.

Reply:
Hmm.. I think I am getting what you are saying.. I tried it out but unfortunately this particular `Entry` itself will have three variants depending on feature list.. And different parent modules are available with different  So there's no way to do it without different feature gating combination at each module we wanna call `Entry` in.. It seems to me this way of consolidating all required imports is better than having various feature gates in different modules.. If later on we want to use both kinda `Entry` together we can explicitly import and give different names to  them here in `collections` module. like `use::std::collections::btree_map::Entry as StdBtreeEntry` or something like that..

Reply:
We can make `hash_brown` a non-optional dependency if that helps.